cmd/skopeo/tags.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/containers/image/docker"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+
+	"io"
+	"strings"
+)
+
+// tagListOutput is the output format of (skopeo tags), primarily so that we can format it with a simple json.MarshalIndent.
+type tagListOutput struct {
+	DockerRepository string   `json:",omitempty"`
+	Tags             []string `json:",omitempty"`
+}
+
+type tagsOptions struct {
+	global *globalOptions
+	image  *imageOptions
+	raw    bool // Output the raw manifest instead of parsing information about the image
+	config bool // Output the raw config blob instead of parsing information about the image
+}
+
+func tagsCmd(global *globalOptions) cli.Command {
+	sharedFlags, sharedOpts := sharedImageFlags()
+	imageFlags, imageOpts := imageFlags(global, sharedOpts, "", "")
+
+	opts := tagsOptions{
+		global: global,
+		image:  imageOpts,
+	}
+
+	return cli.Command{
+		Name:  "docker-list-tags",
+		Usage: "List tags in a Docker v2 API registry and repository specified by the REPOSITORY-NAME",
+		Description: fmt.Sprintf(`
+	Return the list of tags in from the repository in the Docker v2 registry "REPOSITORY-NAME"
+
+	See skopeo(1) section "REPOSITORY NAMES" for the expected format
+	`),
+		ArgsUsage: "REPOSITORY-NAME",
+		Flags:     append(sharedFlags, imageFlags...),
+		Action:    commandAction(opts.run),
+	}
+}
+
+func (opts *tagsOptions) run(args []string, stdout io.Writer) (retErr error) {
+	ctx, cancel := opts.global.commandTimeoutContext()
+	defer cancel()
+
+	if len(args) != 1 {
+		return errors.New("Usage: skopeo docker-list-tags <registry host>:<port>/<repository name>")
+	}
+
+	// Add a leading // to make it valid docker reference, even though user doesn't specify the transport
+	repoRef := fmt.Sprintf("//%s", args[0])
+
+	repo, err := docker.ParseReference(repoRef)
+	if err != nil {
+		return err
+	}
+
+	outputData := tagListOutput{
+		DockerRepository: repo.DockerReference().Name(),
+		Tags:             []string{},
+	}
+
+	sys, err := opts.image.newSystemContext()
+	if err != nil {
+		return err
+	}
+
+	outputData.Tags, err = docker.GetRepositoryTags(ctx, sys, repo)
+
+	if err != nil {
+		// some registries may decide to block the "list all tags" endpoint
+		// gracefully allow the inspect to continue in this case. Currently
+		// the IBM Bluemix container registry has this restriction.
+		if !strings.Contains(err.Error(), "401") {
+			return fmt.Errorf("Error determining repository tags: %v", err)
+		}
+		logrus.Warnf("Registry disallows tag list retrieval; skipping")
+	}
+
+	out, err := json.MarshalIndent(outputData, "", "    ")
```

Comment:
(RFC: Not sure that JSON should be the only possible output format; a simple one-tag-per-line could be simpler to read and consume from shell scripts.

OTOH this works perfectly fine, and other formats can be added as options by whoever needs them.)

Reply:
I kept symmetry with the inspect output for now. Can add non-json output if that's a blocker